settings/urls/v1.py

```diff
@@ -0,0 +1,11 @@
+from django.urls import include, path
+from rest_framework import routers
+from blog.views import PostViewSet, PostDetailViewSet
+
+router = routers.DefaultRouter()
+router.register(r'posts', PostViewSet)
+router.register(r'postsdetails', PostDetailViewSet)
```

Comment:
PostDetails is returning list of posts with details which can't be implemented to view details about a single post. 
Try using POSTMAN to run requests against your API when developing.
For example for posts it gives me:
```
{
    "count": 2,
    "next": null,
    "previous": null,
    "results": [
        {
            "author": {
                "username": "hvitis"
            },
            "title": "Lekcja 1. Wprowadzenie",
            "overview": "gererg",
            "content": "<p>gergerg</p>",
            "publish": "2020-04-20T23:35:27Z",
            "thumbnail": "http://127.0.0.1:8000/media/blog/default.jpeg"
        }
}
```

and for postsdetails
```
{
    "count": 2,
    "next": null,
    "previous": null,
    "results": [
        {
            "id": 2,
            "author": {
                "username": "hvitis"
            },
            "meta_title": null,
            "meta_description": null,
            "meta_keywords": "",
            "title": "Lekcja 1. Wprowadzenie",
            "slug": "2020-4-20-lekcja-1-wprowadzenie",
            "overview": "gererg",
            "content": "<p>gergerg</p>",
            "timestamp": "2020-04-20T23:35:35.680272Z",
            "updated": "2020-04-20T23:35:35.680330Z",
            "status": "published",
            "publish": "2020-04-20T23:35:27Z",
            "thumbnail": "http://127.0.0.1:8000/media/blog/default.jpeg"
        }
}
```

In both cases they return paginated list of 10 posts. The post details should return 1 post depending on the URL query parameter:
e.g.
```
    def get_object(self):
        queryset = []
        try:
            unique_id = generateUUID.UUID(str(self.kwargs["uuid"]))
            queryset = UserProfile.objects.get(unique_id=unique_id)
        except:
            pass
        return queryset
        serializer_class = LocationSerialiazer
```

Try checking up on REST API [good practices ](https://hackernoon.com/restful-api-designing-guidelines-the-best-practices-60e1d954e7c9)

Relevant here is:

The resource should always be plural in the API endpoint and if we want to access one instance of the resource, we can always pass the id in the URL.

> method GET path /companies should get the list of all companies
> method GET path /companies/34 should get the detail of company 34
> method DELETE path /companies/34 should delete company 34